Observer/CheckoutSubmitBefore.php

```diff
@@ -0,0 +1,207 @@
+<?php
+
+/**
+ * PAYONE Magento 2 Connector is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PAYONE Magento 2 Connector is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with PAYONE Magento 2 Connector. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * PHP version 5
+ *
+ * @category  Payone
+ * @package   Payone_Magento2_Plugin
+ * @author    FATCHIP GmbH <support@fatchip.de>
+ * @copyright 2003 - 2016 Payone GmbH
+ * @license   <http://www.gnu.org/licenses/> GNU Lesser General Public License
+ * @link      http://www.payone.de
+ */
+
+namespace Payone\Core\Observer;
+
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Quote\Model\Quote;
+use Magento\Quote\Api\Data\AddressInterface;
+
+/**
+ * Event class to set the orderstatus to new and pending
+ */
+class CheckoutSubmitBefore implements ObserverInterface
+{
+    /**
+     * PAYONE consumerscore request model
+     *
+     * @var \Payone\Core\Model\Api\Request\Consumerscore
+     */
+    protected $consumerscore;
+
+    /**
+     * Consumerscore helper
+     *
+     * @var \Payone\Core\Helper\Consumerscore
+     */
+    protected $consumerscoreHelper;
+
+    /**
+     * Constructor
+     *
+     * @param \Payone\Core\Model\Api\Request\Consumerscore $consumerscore
+     * @param \Payone\Core\Helper\Consumerscore            $consumerscoreHelper
+     */
+    public function __construct(
+        \Payone\Core\Model\Api\Request\Consumerscore $consumerscore,
+        \Payone\Core\Helper\Consumerscore $consumerscoreHelper
+    ) {
+        $this->consumerscore = $consumerscore;
+        $this->consumerscoreHelper = $consumerscoreHelper;
+    }
+
+    /**
+     * Get parameter from config
+     *
+     * @param  string $sParam
+     * @param  bool   $blIsAddresscheck
+     * @return string
+     */
+    protected function getConfigParam($sParam, $blIsAddresscheck = false)
+    {
+        $sGroup = 'creditrating';
+        if ($blIsAddresscheck === true) {
+            $sGroup = 'address_check';
+        }
+        return $this->consumerscoreHelper->getConfigParam($sParam, $sGroup, 'payone_protect');
+    }
+
+    /**
+     * Determine if creditrating is needed
+     *
+     * @param  Quote $oQuote
+     * @return bool
+     */
+    protected function isCreditratingNeeded(Quote $oQuote)
```

Comment:
This method is too complex, please refactor: https://scrutinizer-ci.com/g/PAYONE-GmbH/magento-2/inspections/f57e816f-b8e5-4e29-b0a2-aa3d5399368d/code-structure/operation/Payone%5CCore%5CObserver%5CCheckoutSubmitBefore%3A%3AisCreditratingNeeded